src/hera/workflows/v5/examples/upstream/dag_conditional_parameters.py

```diff
@@ -0,0 +1,64 @@
+from hera.expr import g
+from hera.workflows.v5.dag import DAG
+from hera.workflows.v5.parameter import Parameter
+from hera.workflows.v5.script import Script
+from hera.workflows.v5.task import Task
+from hera.workflows.v5.workflow import Workflow
+
+
+def heads():
+    print("heads")
+
+
+def tails():
+    print("tails")
+
+
+# fmt: off
+def flip_coin():
+    import random
+    print("heads" if random.randint(0,1) == 0 else "tails")
+# fmt: on
+
+
+def get_script(callable):
+    return Script(
+        name=callable.__name__.replace("_", "-"),
+        source=callable,
+        add_cwd_to_sys_path=False,
+        image="python:alpine3.6",
+    )
+
+
+heads_template = get_script(heads)
+tails_template = get_script(tails)
+flip_coin_template = get_script(flip_coin)
+
+with Workflow(
+    generate_name="dag-conditional-parameter-",
+    entrypoint="main",
+    namespace=None,
+    labels={"workflows.argoproj.io/test": "true"},
+    annotations={
+        "workflows.argoproj.io/description": (
+            "Conditional parameters provide a way to choose the output parameters "
+            "based on expression.\n\nIn this example DAG template has two task which"
+            " will run conditionally based on `when`.\n\nBased on this condition one"
+            " of task may not execute."
+            " The template's output parameter will be set to the\nexecuted taks's output result.\n"
```

Comment:
The typos 😭 But they're from the upstream example - are we syncing files down to this repo or copied once to compare python<>yaml?

If they're not resynced:
```suggestion
            "Conditional parameters provide a way to choose the output parameters "
            "based on an expression.\n\nIn this example the DAG template has two tasks which"
            " will run conditionally based on `when`.\n\nBased on this condition, one"
            " of the tasks may not execute."
            " The template's output parameter will be set to the\nexecuted task's output result.\n"
```

Reply:
They are synced unfortunately.